packages/wrangler/src/dev.tsx

```diff
@@ -303,6 +303,15 @@ export function devOptions(yargs: Argv<CommonYargsOptions>): Argv<DevArgs> {
 }
 
 export async function devHandler(args: ArgumentsCamelCase<DevArgs>) {
+	if (!args.local) {
+		const isLoggedIn = await loginOrRefreshIfRequired();
+		if (!isLoggedIn) {
+			throw new Error(
+				"You must be logged in to use wrangler dev in remote mode. Try logging in, or run wrangler dev --local."
+			);
+		}
+	}
+
```

Comment:
Should this be tested or is already?

Reply:
Uhhhh i'm struggling to write the test for this

Reply:
nvm wrote one